rust/README.md

```diff
@@ -13,6 +13,19 @@ Examples
 By default, all example logs into `./log/<example>_rCURRENT.log`
 You can inspect the logs using `tail -F log/its-client_rCURRENT.log`
 
+### json_counter
+
+Subscribes to `test.mosquitto.org` and yields how much messages were received and the number of them that had JSON payload
+
+```
+cargo run --example json_counter
```

Comment:
From what I understand, the warning is telling that your environment is set with `resolver=1` while the project is set using `resolver=2`
This is because `edition=2021` is set in Cargo.toml and is automatically sets resolver to 2

> The version "1" resolver is the original resolver that shipped with Cargo up to version 1.50. The default is "2" if the root package specifies [edition = "2021"](https://doc.rust-lang.org/cargo/reference/manifest.html#the-edition-field) or a newer edition. Otherwise the default is "1".

From the [Cargo book](https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions)

You need to check your environment edition and versions